[ADD] accounting/l10n_uy: uruguay

samueljlieber opened the pull request:
Task: [#3997556](https://www.odoo.com/odoo/project/3835/tasks/3997556?cids=3)

Introduce Uruguay fiscal localization.​

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+
+The supported documents are:
+
+- Invoices, credit notes, debit notes;
```
```suggestion
- e-Invoice, e-Invoice Credit Notes, e-Invoice Debit Notes;
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+The supported documents are:
+
+- Invoices, credit notes, debit notes;
+- e-ticket e-ticket credit notes, e-ticket debit notes;
```
```suggestion
- e-Ticket, e-Ticket Credit Notes, e-Ticket Debit Notes;
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+
+- Invoices, credit notes, debit notes;
+- e-ticket e-ticket credit notes, e-ticket debit notes;
+- Export e-invoice, export e-credit notes, export e-debit notes.
```
```suggestion
- Export e-Invoice, Export e-Invoice Credit Note, Export e-Invoice Debit Note.
```

fvz-odoo replied:
Added this comments to be consistent on how the documents are named in the database.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+
+The following terms are used throughout the Uruguayan localization:
+
+- **DGI**: *Dirección General Impositiva* is the government organization that enforces pay of
+  taxes in Uruguay.
```
```suggestion
- **DGI**: *Dirección General Impositiva* is the government entity responsible for enforcing tax
  payments in Uruguay.
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+       characteristics for the Uruguayan localization, which represent the minimum configuration
+       required for a company to operate in Uruguay according to the guidelines set by the |DGI|.
+       The module's installation automatically loads: chart of accounts, taxes, documents types, and
+       tax support types.
```
```suggestion
       tax supported types.
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+     - Includes all the technical and functional requirements to generate and validate
+       :doc:`Electronics Documents <../accounting/customer_invoices/electronic_invoicing>`, based on
+       the technical documentation published by the |DGI|. The authorized documents are: invoices,
+       credit notes and debit notes.
```
Should we mention the other documents like before (lines 17-19) or should we leave these as example. I suggest to, at least, refer to the previous documents mentioned above.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+
+.. note::
+   Odoo automatically installs the base module *Uruguay - Accounting* when a database is
+   installed with `Uruguay` selected as the country.
```
We should mention that they would need to install the l10n_uy_edi afterwards (if they want to have electronic invoicing, of course).

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+  - :guilabel:`Country`
+
+- :guilabel:`Tax ID`: enter the identification number for the selected taxpayer type.
+- :guilabel:`DGI Main Branch Code`:
```
Brief description of what this field is and point out to get it from Uruware.

rps-odoo replied:
Please add:

- :guilabel:'DGI Main Branch Code': This value is part of the XML when creating an electronic document. If not set, all electronic documents will be rejected. To get this number, follow these steps:
1. Go to 'Servicios en línea DGI' page.
2. Select option 'Registro único tributario -> Consulta de datos- (menu link at the right side).
3. Select the option 'Consulta de Datos Registrales -> Consulta de Datos de Entidades'.
4. Open the generated PDF. Get the number on section 'Domicilio Fiscal -> Número de Local'.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+
+To generate electronic documents in Odoo, a Uruware account is necessary.
+
+To set up a Uruware account, first be sure to have a valid Odoo subscription. Then follow the steps
```
```suggestion
To set up a Uruware account, first be sure to have a valid Odoo subscription and the data of the company correctly filled in. Then follow the steps
```

Not sure if format-wise my suggestion will work, you can rephrase it, but be sure to add that info, the company's information has to be complete.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+
+.. _l10n_uy/uruware-account:
+
+Set up Uruware account
+----------------------
+
+To generate electronic documents in Odoo, a Uruware account is necessary.
+
+To set up a Uruware account, first be sure to have a valid Odoo subscription. Then follow the steps
+proceeding steps:
+
+Locate the Uruware credentials settings by navigating to the :menuselection:`Accounting -->
+Configuration --> Settings: Uruguay Localization section`. Take a note of the admin email address
+that is set, this will be the same email used in Uruware. Click on :guilabel:`Create Uruware
+Account`.
+
+.. note::
+   This action will create an account with Uruware with the following information:
+
+   - Legal name (razón social)
+   - RUT from the company
+   - Administrator email
+   - User name and last name
+   - Odoo database link
+
+.. important::
+   Add in any of the information that is missing to create your account.
+
+Once the account is created, a confirmation email is sent to the email used to create the account,
+containing the account credentials to configure the account directly in the Uruware portal:
+
+- Testing portal: https://odootest.ucfe.com.uy/Gestion/
+- Production portal: https://prod6109.ucfe.com.uy/Gestion/
+
+Use the account credentials in the email to log in to the portal.
```
We should add a comment somehow (could be as an important note) that they will need to create a testing account and a production account. And that they just can create the account once.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+   :align: center
+
+.. note::
+   These credentials can be obtained from the email received when the :ref:`Uruguay account was
+   created <l10n_uy/uruware-account>`.
```
Not really, and actually we should be careful, the credentials they receive in the email will be to configure their account in Uruware, and in Uruware they will get these credentials (Uruware WS password, commerce code and terminal code).

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+Some accounting transactions like *Customer Invoices* and *Vendor Bills* are classified by document
+types. These are defined by the government fiscal authorities, in this case by the |DGI|.
+
+Each document type can have a unique sequence per journal where it is assigned. The data is created
+automatically when the localization module is installed, and the information required for the
+document types is included by default.
```
Although this is true, we should add a note to specify that in Uruguay CAE are needed and have to be uploaded in Uruware. In Odoo we will get the sequence from Uruware based on their CAEs (works like in Chile, but in Uruguay they need to upload them in Uruware). CAEs are only used in production, in test they just need to set the range of sequences to be used in Uruware.

rps-odoo replied:
We should add the following note @samueljlieber 

.. note::
   It is necessary to upload the document type sequence (known as CAE) in the Uruware portal. Odoo will get the sequence and PDF file from Uruware based on the CAEs. CAEs are only used in production. For the testing environment, a set range of sequences is needed. This is set in the testing portal from Uruware.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+To generate and confirm an electronic document that will be validated by |DGI|, the sales journal
+needs to be configured with the following:
+
+- :guilabel:`Invoicing Type`: manual invoices are for open invoices previously stamped in another
```
The electronic option is missing here with its explanation. And electronic will be set by default.

rps-odoo replied:
Add following bullet point @samueljlieber

- :guilabel:`Ecuadorina Edi format`: This checkbox will be active by default. This is necessary to send electronic documents via web service to the Ecuadorian government.

fvz-odoo replied:
> Add following bullet point @samueljlieber
> 
> * :guilabel:`Ecuadorian edi format`: This checkbox will be active by default. This is necessary to send electronic documents via web service to the Ecuadorian government.

@rps-odoo Ecuadorian? 🤔

rps-odoo replied:
@fvz-odoo don't ask me, ask Odoo
![Screenshot 2024-07-08 at 2 31 32 PM](https://github.com/odoo/documentation/assets/79879654/57ccf18a-1fa4-4b20-a794-5b7757e78e37)

fvz-odoo replied:
> @fvz-odoo don't ask me, ask Odoo. I will change it to Spanish.
> 
> ![Screenshot 2024-07-08 at 2 31 32 PM](https://private-user-images.githubusercontent.com/79879654/346716551-57ccf18a-1fa4-4b20-a794-5b7757e78e37.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0NzI3ODQsIm5iZiI6MTcyMDQ3MjQ4NCwicGF0aCI6Ii83OTg3OTY1NC8zNDY3MTY1NTEtNTdjY2YxOGEtMWZhNC00YjIwLWE3OTQtNWI3NzU3ZTc4ZTM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA4VDIxMDEyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk1YjBlNThhYzg5ODAwNzMzZGI2Y2FkMzVlOWY2Y2E4YTRiZjczZjkyYmQ3ODAyNWUwYjUzZjJmMzI2ZmQ3YmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.iqixzT38eeGnn2M2y_J_A8UEXxADnghutIdT07yogqo)

Hello @rps-odoo 

Mmmm, I don't understand why we have this setting in the Uruguayan localization. We're probably talking about different things? 

In the Uruguayan localization we have a selection field (l10n_uy_edi_type):
![Screenshot 2024-07-08 at 2 08 27 PM](https://github.com/odoo/documentation/assets/75649179/f4e36d99-052f-44d8-80e5-28feff7b43e3)

If, for some reason, we have that Ecuadorian field in our module, we have something wrong, but I don't see it. Can you please confirm we have it and that we're talking about the same thing?

Let me know.

Cheers!

Fernando

@samueljlieber

rps-odoo replied:
@fvz-odoo Ops, I am confusing myself. I am also working on another PR for Ecuador and got confused. Don't worry,  everything seems as you explain. @samueljlieber please ignore all these comments and just add the following:

:guilabel:'Invoicing Type': This option will have 'Electronic' value set by default. This is necessary to send electronic documents via web service to the Ecuadorian government.

fvz-odoo replied:
> @fvz-odoo Ops, I am confusing myself. I am also working on another PR for Ecuador and got confused. Don't worry, everything seems as you explain. @samueljlieber please ignore all these comments and just add the following:
> 
> * :guilabel:'Invoicing Type': This option will have 'Electronic' value set by default. This is necessary to send electronic documents via web service to the Ecuadorian government.

One thing @samueljlieber, please, just make sure it says **Uruguayan** government at the end 😅.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+needs to be configured with the following:
+
+- :guilabel:`Invoicing Type`: manual invoices are for open invoices previously stamped in another
+  system, for example directly in the |DGI|. Electronic invoices will not be sent to Uruware.
```
This is a little confusing, because it says electronic invoices will not be sent to Uruware, but if they select "electronic" they will be sent to Uruware, maybe rephrase this paragraph to explain the manual option with "manual invoices are for open invoices previously stamped in another system, for example in the DGI" and then explain the electronic option.

rps-odoo replied:
Please rephrase bullet point @samueljlieber 

- :guilabel:'Invoicing Type': This option will have 'Electronic' value set by default. This is necessary to send electronic documents via web service to the Uruguayan government. The other option ('Manual') is for open invoices previously stamped in another system, for example, in the DGI.

fvz-odoo replied:
@samueljlieber somehow this was shown as resolved but it wasn't, can you please rephrase to:

:guilabel:'Invoicing Type': This option will have 'Electronic' value set by default. This is necessary to send electronic documents via web service to the Uruguayan government through Uruware. The other option ('Manual') is for open invoices previously stamped in another system, for example, in the DGI.

fvz-odoo replied:
This one also @samueljlieber can you please check? 😉

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+
+- :guilabel:`Invoicing Type`: manual invoices are for open invoices previously stamped in another
+  system, for example directly in the |DGI|. Electronic invoices will not be sent to Uruware.
+- :guilabel:`Use Documents?`: Activate this option if this journal will be used to generate
+  electronic documents and sent to the government through Uruware.
```
```suggestion
- :guilabel:`Use Documents?`: Activate this option if this journal will use documents from the list of document types in Odoo.
```

This boolean is to be able to select document types in account.move, but you can also select it if the journal is manual (example, a journal used for open invoices with different document types).

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+- :guilabel:`Customer`: type the customer's information.
+- :guilabel:`Due date`: to compute if the invoice is due now or later (contado or crédito
+  respectively).
+- :guilabel:`Journal`: select the sales electronic journal.
```
```suggestion
- :guilabel:`Journal`: select the electronic sales journal.
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+- :guilabel:`Due date`: to compute if the invoice is due now or later (contado or crédito
+  respectively).
+- :guilabel:`Journal`: select the sales electronic journal.
+- :guilabel:`Document Type`: type document type in this format, for example, `(111) e-Invoice`.
```
```suggestion
- :guilabel:`Document Type`: document type in this format, for example, `(111) e-Invoice`.
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+The :doc:`Customer credit note <../accounting/customer_invoices/credit_notes>` is an electronic
+document that, when validated, is sent to |DGI| via Uruware. It is necessary to have a validated
+(posted) invoice to register a credit note. On the invoice there is a button named :guilabel:`Credit
+note`, click on this button to be directed to the :guilabel:`Create credit note` form, then complete
```
```suggestion
note`, click this button to be directed to the :guilabel:`Create credit note` form, then complete
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+  Documents?` option active.
+- :guilabel:`Document Type`: select the credit note document type.
+- :guilabel:`Reversal Date`: type the date.
+
```
Add a note here that every document type has its specific credit note and debit note, example e-Invoice > e-Invoice credit Note or e-Ticket > e-Ticket Credit Note.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+  Documents?` option active.
+- :guilabel:`Copy lines`: mark checkbox to copy the invoice lines to the debit note.
+- :guilabel:`Debit note date`: type the date.
+
```
Maybe add [this note](https://github.com/odoo/documentation/pull/9849/files#r1653641712) here, since it refers to both credit and debit notes.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+can be mandatory or optional. To create a new addenda, go to :menuselection:`Accounting --> Settings
+--> DGI --> Addendas and disclosures`.
+
+To create an addenda, the following information is required:
+
+- :guilabel:`Name`: Name of the Addenda or Mandatory Disclosure.
+- :guilabel:`Apply on`: Select the documents where the addenda or mandatory disclosure are
+  applicable.
+- :guilabel:`Type`: Select the type of remark, this will add it to the specific section in the XML.
+- :guilabel:`Content`: Add the complete text of the addenda or disclosure.
+- :guilabel:`Is disclosure?`: Select this box if the text is a mandatory disclosure, leave it blank
+  if it is additional information.
```
We removed the "Apply on" filed. Let's remove it from here too. Also the "Is disclosure" field changed to "Is Legend" and goes before the content.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,321 @@
+preconfigured addenda and disclosure to the product in the invoice line. The field called
+:guilabel:`Disclosure` will be used to add the leyenda to the product specified in the line.
+
+Leyenda and additional information in document
+----------------------------------------------
+
+To add a leyenda or additional information to the electronic document and XML, it is necessary to
+add the preconfigured addenda and disclosure to the document using the field :guilabel:`Addenda and
+Disclosures` that can be found in the *Other Info* tab in the document form view. The addenda and
+disclosures added here will appear in the XML and visibly in the PDF document.
```
This applies for leyendas and additional information in documents + issuer, receiver and actual addendas. Just expand the explanation to all 4 of them.

rps-odoo replied:
@samueljlieber rephrase section with:

samueljlieber commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,320 @@
+Review taxes
+~~~~~~~~~~~~
+
+As part of the Uruguay localization module, taxes are automatically created with its configuration
+and related financial accounts.
```
Need navigation instructions to where taxes can be reviewed.

rps-odoo replied:
Add the following:

To review these taxes, go to :menuselection:`Accounting --> Configurations --> Taxes`.

samueljlieber commented:
Hi @fvz-odoo, thank you so much for your helpful review! I implemented most of your suggestions in fcc950e, as well as added missing images, and added a couple of suggestions of my own.

@rps-odoo can you help me with the following:
- [DGI Main Branch Code description](https://github.com/odoo/documentation/pull/9849#discussion_r1653606131)
- [Specify that in Uruguay CAEs are needed and must be uploaded in Uruware](https://github.com/odoo/documentation/pull/9849#discussion_r1653627741)
- [Electronic option is missing here with its explanation](https://github.com/odoo/documentation/pull/9849#discussion_r1653629338)
- [Rephrase this paragraph to explain the manual option](https://github.com/odoo/documentation/pull/9849#discussion_r1653633818)
- [Expand the explanation to all 4](https://github.com/odoo/documentation/pull/9849#discussion_r1653646999)
- [Need navigation instructions for electronic invoice](https://github.com/odoo/documentation/pull/9849#discussion_r1662581670)
- [Need navigation instructions to where taxes can be reviewed](https://github.com/odoo/documentation/pull/9849#discussion_r1662683070)
- [Need navigation instructions for where can users can manage document types](https://github.com/odoo/documentation/pull/9849#discussion_r1662684275)

Thank you!

rps-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,320 @@
+-----------------------
+
+To configure the electronic invoice data, an environment and credentials need to be configured. To
+do so, navigate to :menuselection:`... --> ...`.
```
:menuselection:`Accounting --> Configurations --> Settings --> Localización Ecuatoriana`.

rps-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,320 @@
+Some accounting transactions like *Customer Invoices* and *Vendor Bills* are classified by document
+types. These are defined by the government fiscal authorities, in this case by the |DGI|.
+
+Each document type can have a unique sequence per journal where it is assigned. The data is created
+automatically when the localization module is installed, and the information required for the
+document types is included by default.
```
Add the following:

To review these taxes, go to :menuselection:`Accounting --> Configurations --> Document Types`.

fvz-odoo replied:
> Add the following:
> 
> To review these taxes, go to :menuselection:`Accounting --> Configurations --> Document Types`.

"To review the document types included in the localization:`Accounting --> Configurations --> Document Types`"

samueljlieber commented:
Thanks all! I implemented all the [comments](https://github.com/odoo/documentation/pull/9849#issuecomment-2203495234) in 41846d6.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,354 @@
+
+To generate electronic documents in Odoo, a Uruware account is necessary.
+
+To set up a Uruware account, first be sure to have a valid Odoo subscription. Then follow the steps
+proceeding steps:
```
```suggestion
To set up a Uruware account, first be sure to have a valid Odoo subscription. Then follow the
proceeding steps:
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,354 @@
+
+Locate the Uruware credentials settings by navigating to the :menuselection:`Accounting -->
+Configurations --> Settings --> Localización Uruguaya`. Take a note of the admin email address that
+is set, this will be the same email used in Uruware. Click on :guilabel:`Create Uruware Account`.
```
```suggestion
is set, this will be the same email used in Uruware's portal. Click on :guilabel:`Create Uruware Account`.
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,354 @@
+   - User name and last name
+   - Odoo database link
+
+   Add in any of the information that is missing to create your account.
```
```suggestion
   To ensure your account is created correctly, please add any missing information from above.
```

samueljlieber commented:
Thanks @fvz-odoo! I implemented your additional changes in f5c1c6d

samueljlieber commented:
Hi @ewa-odoo, requesting your review before opening this PR up from draft, thank you!

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,355 @@
+
+  To find the *DGI Main Branch Code*, follow these steps:
+
+  #. From your :ref:`Uruware account <l10n_uy/uruware-account>`, go to :menuselection:`Servicios en
+     línea DGI --> Registro único tributario --> Consulta de datos`.
```
Hello @samueljlieber and @ewa-odoo 

Sorry, checking this part, what we're pointing is to the reference from the DGI (and it is correct)... But there are 2 options here, change the "ref: 'Uruware account' to DGI or, the instructions change them to the way they should find the Branch code in Uruware. Both are correct. My opinions is to the first (change to DGI) since at this point the Uruware account has not been created.

Let us know please.

Thanks!

fernando

@marianaluz

fvz-odoo replied:
Hey @samueljlieber 

This one was pending, as I can see

```suggestion
  #. From your :ref:`DGI account <https://servicios.dgi.gub.uy/serviciosenlinea>`, go to :menuselection:`Servicios en
     línea DGI --> Registro único tributario --> Consulta de datos`.
```

Not sure if this is the way to put in the DGI website, but you can add it the right way ;).

Cheers!

Fernando

fvz-odoo replied:
Hi @samueljlieber I don't see this change yet, could you please check? 😬

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,355 @@
+
+Locate the Uruware credentials settings by navigating to the :menuselection:`Accounting -->
+Configurations --> Settings --> Localización Uruguaya`. Take a note of the admin email address that
+is set, this will be the same email used in Uruware's portal. Click on :guilabel:`Create Uruware
```
Hello @samueljlieber and @ewa-odoo 

I suggest to add a tip box with the following "Enter the email of the administrator or project leader. Avoid using the same email as the one used for communication with the DGI" or add it in parenthesis after email used in Uruware's portal, this way users will know which email to use easier.

Cheers!

Fernando

@marianaluztorres

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,355 @@
+-----------------------
+
+To configure the electronic invoice data, an environment and credentials need to be configured. To
+do so, navigate to :menuselection:`Accounting --> Configurations --> Taxes`.
```
```suggestion
do so, navigate to :menuselection:`Accounting --> Configurations --> Uruguayan Localization`.
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,355 @@
+   :align: center
+
+.. note::
+   These credentials can be obtained from the Uruware portal, after configuring the :ref:`Uruware
+   account <l10n_uy/uruware-account>`.
```
```suggestion
   account <l10n_uy/uruware-account>`. The WS Password is different from the password you use to enter your Uruware account.
```

fvz-odoo replied:
```suggestion
   This data can be obtained from the Uruware portal, after configuring the :ref:`Uruware
   account <l10n_uy/uruware-account>`. The WS Password is different from the password you use to enter your Uruware account.
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,355 @@
+  to the |DGI|. For this reason, rejection errors will not appear in this mode. Every internal
+  validation can be tested in demo mode. Avoid selecting this option in a production database.
+
+Then, enter the :guilabel:`Uruware Credentials`:
```
```suggestion
Then, enter the :guilabel:`Uruware Data`:
```

samueljlieber commented:
Thanks @fvz-odoo! I implemented each of your resolved suggestions in https://github.com/odoo/documentation/commit/ee4c2d94dfcedee1b4572c5ad9a3385297457e93, thank you! 

@gmz-odoo can you help me with this remaining suggestion: https://github.com/odoo/documentation/pull/9849#discussion_r1714242710 🙏

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,354 @@
+Use the account credentials in the email to log in to the portal.
+
+.. important::
+   Be sure to configure a testing account and a production account in the Uruware portal. The same
+   Uruware account can be used for both environments.
```
```suggestion
   Be sure to configure two accounts, one for testing and one for production. The email address 
   can be changed in Uruware and the same can be used in both environments.
```

samueljlieber replied:
https://github.com/odoo/documentation/pull/9849#discussion_r1715487691 implemented in 7d28b87

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,354 @@
+
+To generate electronic documents in Odoo, a Uruware account is necessary.
+
+To set up a Uruware account, first be sure to have a valid Odoo subscription. Then follow the
+proceeding steps:
+
+Locate the Uruware credentials settings by navigating to the :menuselection:`Accounting -->
+Configurations --> Settings --> Localización Uruguaya`. Take a note of the admin email address that
+is set, this will be the same email used in Uruware's portal. Click on :guilabel:`Create Uruware
+Account`.
+
+.. tip::
+   Enter the email of the administrator or project leader. Avoid using the same email as the one
+   used for communication with the |DGI|.
```
```suggestion
To set up a Uruware account, first be sure to have a valid Odoo subscription. Then follow the
proceeding steps:

Locate the Uruware credentials settings by navigating to the :menuselection:`Accounting -->
Configurations --> Settings --> Localización Uruguaya`. Click on :guilabel:`Create Uruware
Account`. You will receive an email to the address associated with your Odoo subscription with the  
username and password to enter Uruware's portal and setup your account.

.. tip::
   Your  :guilabel:`Tax ID` needs to be setup to be able to create your account.
```

Sorry for this last minute change @samueljlieber but the "admin email" field will no longer be available. The email that will be used will be the one associated to the customers's Odoo subscription for security reasons.

samueljlieber commented:
Thanks @fvz-odoo! Implemented in https://github.com/odoo/documentation/pull/9849#discussion_r1725467858

fvz-odoo commented:
> Thanks @fvz-odoo! Implemented in [#9849 (comment)](https://github.com/odoo/documentation/pull/9849#discussion_r1725467858)

Thanks @samueljlieber this LGTM now. We can merge this once the [edi module](https://github.com/odoo/enterprise/pull/51266) gets merged, thanks!

Cheers!

Fernando

@ewa-odoo

samueljlieber commented:
https://github.com/odoo/documentation/commit/4d56fdc8b5f1ed983be7364849482d3a12735fb4 Rebased to latest and opened this PR up for review!

@odoo/localizations-doc-review this PR is ready for your review! Please consider this PR a priority as the localization is going live this week! Thanks in advance!

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+- :guilabel:`Commerce Code`
+- :guilabel:`Terminal Code`
+
+.. image:: uruguay/electronic-invoice-data.png
```
Hello @samueljlieber 

One quick change since I told you the change in the text but didn't see the image. Can you change the above image to this one (will send you the file through discord):

![Screenshot 2024-09-05 at 2 13 22 PM](https://github.com/user-attachments/assets/35673177-ed72-493f-a05f-ba299a038c59)

Let me know if anything!

Thanks!

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+
+   - Legal name (razón social)
+   - RUT from the company
+   - Administrator email
+   - User name and last name
```
```suggestion
   - User name will be your RUT.odoo
```
@samueljlieber

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+
+   To ensure your account is created correctly, please add any missing information from above.
+
+Once the account is created, a confirmation email is sent to the email used to create the account,
+containing the account credentials to configure the account directly in the Uruware portal:
```
```suggestion
Once the account is created, a confirmation email is sent to the email linked to your subscription in Odoo,
containing the credentials to configure the account directly in the Uruware portal:
```
@samueljlieber

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+as part of the set of data included in the localization module, the accounts are mapped
+automatically in taxes, default accounts payable, and default accounts receivable.
+
+Accounts can by added or deleted according to the company's needs.
```
```suggestion
Accounts can be added or deleted according to the company's needs.
```
@samueljlieber

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+
+- :guilabel:`Identification Number`:
+
+  - :guilabel:`Type`: select a identification number type.
```
```suggestion
  - :guilabel:`Type`: select a identification type.
```
@samueljlieber

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+- :guilabel:`Products`: specify the product with the correct taxes.
+
+.. note::
+   Every document type has a specific credit note and debit note (e.g., the document type *(111)
+   e-Invoice* has an *e-Invoice* credit note).
```
```suggestion
   Every document type has a specific credit note and debit note (e.g., the document type *(111)
   e-Invoice* has an *(112) e-Invoice Credit Note*).
```
@samueljlieber

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+- :guilabel:`Copy lines`: mark checkbox to copy the invoice lines to the debit note.
+- :guilabel:`Debit note date`: type the date.
+
+.. note::
+   The document sequence (number) is brought from Uruware once the document has been processed.
```
Let's add something important here, please, @samueljlieber:

```suggestion
.. note::
    Confirm the invoice and it will be created with an internal reference. To send the document to DGI via Uruware,
    click on "Send and Print" and select the checkbox "Create CFE". The legal document sequence (number) is
    brought from Uruware once the document has been processed. Make sure you have CAEs available in Uruware.
```

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+- Document
+- Issuer
+- Receiver
+- Addendas
```
We can also add at the end, please, @samueljlieber:

.. note::
   To add a temporary note to the electronic document, use the Terms and Conditions field, this information will be
   sent in the addenda of the invoice, but it won't be saved for future documents.

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,350 @@
+  payments in Uruguay.
+- **EDI**: *Electronic Data Interchange* refers to the sending of electronic documents.
+- **Uruware**: is the third-party organization that facilitates the interchange of electronic
+  documents between companies and the Uruguayan government.
+
```
We can add:

- **CAE**: *Constancia de Autorización de Emisión* is a document requested from the tax authority's website to
   enable electronic invoice issuance.

samueljlieber commented:
https://github.com/odoo/documentation/commit/bfb0940ecd84623ace1dae7a821c78e22c55615d Implemented @fvz-odoo final changes :)

dade-odoo commented:
@samueljlieber Hey Sam, thanks for reaching out. I'm passing this straight on to the final review from @auva-odoo to hopefully get it live by EOD Monday :)

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,360 @@
+automatically when the localization module is installed, and the information required for the
+document types is included by default.
+
+To review the document types included in the localization, navigate to :menuselection:`Accounting
+--> Configurations --> Document Types`.
```
```suggestion
To review the document types included in the localization, navigate to :menuselection:`Accounting
--> Configuration --> Document Types`.
```

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,360 @@
+following data:
+
+- :guilabel:`Customer`: type the customer's information.
+- :guilabel:`Due date`: to compute if the invoice is due now or later (contado or crédito
+  respectively).
```
```suggestion
- :guilabel:`Due date`: to compute if the invoice is due now or later (*contado* or *crédito*,
  respectively).
```

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,360 @@
+- :guilabel:`Products`: specify the product with the correct taxes.
+
+.. note::
+   Every document type has a specific credit note and debit note (e.g., the document type *(111)
+   e-Invoice* has an *(112) e-Invoice Credit Note*).
```
```suggestion
   Every document type has a specific credit note and debit note (e.g., the document type :guilabel:`(111)
   e-Invoice` has an :guilabel:`(112) e-Invoice Credit Note`).
```

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,360 @@
+Customer invoices
+~~~~~~~~~~~~~~~~~
+
+:guilabel:`Customer invoices` are electronic documents that, when validated, are sent to |DGI| via
```
```suggestion
Customer invoices are electronic documents that, when validated, are sent to |DGI| via
```
I would not use guilabel here since we're talking about the general term/concept. I would, however, add a reference to the relevant doc.

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,360 @@
+  respectively).
+- :guilabel:`Journal`: select the electronic sales journal.
+- :guilabel:`Document Type`: document type in this format, for example, `(111) e-Invoice`.
+- :guilabel:`Products`: specify the product with the correct taxes.
```
```suggestion
- :guilabel:`Products`: specify the product(s) with the correct taxes.
```

auva-odoo reviewed (commented):
@samueljlieber sorry I clicked Finish review by mistake, I still have a few more lines to go through! Here are a few thoughts already:
- First of all, thank you very much for your work, everything already looks clean and well structured.
- I wasn't able to test everything for some reason and since this is a bit time-sensitive, I focused on what I *could* test and just proofread the rest. We can always go back and change things later if needed.
- For consistency, I’d suggest rephrasing the headings that include a verb, like *Configure your company* to just *Company* (since we’re in the *Configuration* section, we can skip repeating that word in the sub-sections), and *Set a Uruware account* to just *Uruware account*, etc.
- I would also capitalize the first letter after a semi-colon if it's a full sentence (cf the bullet list items in the *Electronic Invoice data* and *Sales documents sections*, for example)
- I noticed thare are still a few unresolved comments from other reviewers.

samueljlieber commented:
Thanks @dade-odoo and @auva-odoo! https://github.com/odoo/documentation/commit/53653accbc86cccfd27833b1dd3ff1d58ee19eb8 I implemented two more very small changes from @fvz-odoo :
- https://github.com/odoo/documentation/pull/9849/files#r1746330094
- https://github.com/odoo/documentation/pull/9849/files#r1746330727

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,360 @@
+
+The supported documents are:
+
+- e-Invoice, e-Invoice Credit Note, e-Invoice Debit Note;
+- e-Ticket, e-Ticket Credit Note, e-Ticket Debit Note;
+- Export e-Invoice, Export e-Invoice Credit Note, Export e-Invoice Debit Note.
```
Hello @auva-odoo 

Thanks for the review, but here it actually makes sense to have them capitalized, because each is a type of document and that's the way they are called in the database. This time we did it on purpose 😉. 

Cheers!

Fernando 

@samueljlieber

auva-odoo replied:
Then in this case we should use the :guilabel: formatting too :slightly_smiling_face:

fvz-odoo replied:
Agreed! 😉

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,360 @@
+
+.. _l10n_uy/uruware-account:
+
+Set up Uruware account
```
Maybe "an Uruware", @auva-odoo ? 

@samueljlieber

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,362 @@
+========================
+
+*Addendas* and *disclosures* are additional notes and comments added to an electronic document that
+can be mandatory or optional. To create a new addenda, go to :menuselection:`Accounting --> Settings
+--> DGI --> Addendas and disclosures`.
```
```suggestion
can be mandatory or optional. To create a new addenda, go to :menuselection:`Accounting --> Configuration --> Settings --> Addendas and disclosures` and click :guilabel:`New`.
```

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,362 @@
+can be mandatory or optional. To create a new addenda, go to :menuselection:`Accounting --> Settings
+--> DGI --> Addendas and disclosures`.
+
+To create an addenda, the following information is required:
```
```suggestion
Enter the the following information:
```
so we don't repeat *To create an addenda*

fvz-odoo replied:
Just remove one of the "the"s 😅, thanks @samueljlieber

fvz-odoo replied:
No worries! Thanks a lot! 😉

fvz-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,360 @@
+Set up Uruware account
+----------------------
+
+To set up a Uruware account, first be sure to have a valid Odoo subscription. Then follow the
```
Yep, this is only for enterprise users only. Community users will not be able to create an account in Uruware. (We pay for each account)

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,362 @@
+---------------------------------------------
+
+To add a leyenda or additional information to the product and XML, it is necessary to add the
+preconfigured addenda and disclosure to the product in the invoice line. The field called
+:guilabel:`Disclosure` will be used to add the leyenda to the product specified in the line.
```
Where is this field exactly? 
```suggestion
preconfigured addenda and disclosure to the product in the invoice line. Add the leyenda in the 
:guilabel:`Disclosure` field of the product specified in the line.
```
(not sure since I don't know where this field is, but I would avoid the phrasing "The field called XXX will be used" in any case :slightly_smiling_face: )

fvz-odoo replied:
The field is in the product line of the invoice. I agree with your suggestion @auva-odoo. 

@samueljlieber

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,362 @@
+Leyenda and additional information
+----------------------------------
+
+To add a leyenda or additional information to the electronic document and XML, it is necessary to
+add the preconfigured addenda and disclosure to the document using the field :guilabel:`Addenda and
+Disclosures` that can be found in the *Other Info* tab in the document form view. The addenda and
```
```suggestion
To add a *leyenda* or additional information to the electronic invoice and XML, access the invoice, go to the :guilabel:`Other Info` tab, and select the desired addenda in the :guilabel:`Addenda and
Disclosure`  field.
```
- It was a bit confusing for me to use *document* insterad of *invoice* here.
- To simplify the sentence

fvz-odoo replied:
Makes sense @auva-odoo (we thought of document because it is also available in other electronic documents, but we were talking about invoices anyway). 

@samueljlieber

samueljlieber commented:
Thank you so much @auva-odoo for your helpful review! I implemented all of your current suggestions in https://github.com/odoo/documentation/commit/a507a526b7737198df4f05a559f9381662bdfc58, I will keep an eye out if any more come through!

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,357 @@
+- Addendas
+
+.. note::
+   To add a temporary note to the electronic document, use the *Terms and Conditions* field, this
+   information will be sent in the addenda of the invoice, but it won't be saved for future
+   documents.
```
```suggestion
   To add a temporary note to the electronic document, use the :guilabel:`Terms and Conditions` field. This
   information will be sent in the addenda of the invoice, but it won't be saved for future documents.

```

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,357 @@
+
+The :doc:`Customer credit note <../accounting/customer_invoices/credit_notes>` is an electronic
+document that, when validated, is sent to |DGI| via Uruware. It is necessary to have a validated
+(posted) invoice to register a credit note. On the invoice there is a button named :guilabel:`Credit
+note`, click this button to be directed to the :guilabel:`Create credit note` form, then complete
```
```suggestion
(posted) invoice to register a credit note. On the invoice, click the :guilabel:`Credit note` button to access the :guilabel:`Create credit note` form, then complete the following information:

```

auva-odoo commented on content/applications/finance/fiscal_localizations/uruguay.rst:
```diff
@@ -0,0 +1,357 @@
+
+The :doc:`Customer debit note <../accounting/customer_invoices/credit_notes>` is an electronic
+document that, when validated, is sent to |DGI| via Uruware. It is necessary to have a validated
+(posted) invoice to register a debit note. On the action within an invoice there is a option named
+:guilabel:`Debit note`, click on this action to be directed to the :guilabel:`Create credit note`
```
```suggestion
(posted) invoice to register a debit note. On the invoice, click the :icon:`cog` icon (:guilabel:`action menu`), select the :guilabel:`Debit note` option to access the :guilabel:`Create credit note`
```

auva-odoo reviewed (approved):
Hey @samueljlieber I think I'm done now :slightly_smiling_face: thank you for your patience! I'm going to go ahead and approve this already since I'm off on Monday. You can r+ when you're ready. Thanks again!

samueljlieber commented:
Thank you so much @auva-odoo for your thorough and very helpful feedback! I implemented your final changes in https://github.com/odoo/documentation/commit/72a3fda61e3f57a057df5d2775a88f52ace4cb9f. Once the checks pass I will merge this PR! 🚀

samueljlieber commented:
@robodoo r+
